src/sentry/static/sentry/app/components/smartSearchBar/utils.tsx

```diff
@@ -72,7 +72,9 @@ export function createSearchGroups(
   const activeSearchItem = 0;
 
   if (maxSearchItems && maxSearchItems > 0) {
-    searchItems = searchItems.slice(0, maxSearchItems);
+    searchItems = searchItems.filter(
+      (value: SearchItem, index: number) => index < maxSearchItems || value.predefined
```

Comment:
Won't this also show additional results for the filters like `assigned` and `firstRelease`?

Reply:
For `firstRelease`, I believe it won't since since `getPredefinedTagValues` will enumerate `Tag.values`...  which is `['latest']`(defined in `src/sentry/static/sentry/app/stores/tagStore.tsx`). The fetch of actual releases happens in `src/sentry/static/sentry/app/components/smartSearchBar/index.tsx:getReleases` where it extends the predefined values with the queried ones (and thus wouldn't have `{predefined: true}`).

For `assigned`... I think you're right that `src/sentry/static/sentry/app/utils/withIssueTags.tsx` will load in more values and set it on `Tag.values` 😭 .

For my strategy to work then... I'd need to replace `Tag.values`currently `Array<string>` with something like `Array<{value: string, predefined: boolean>` so I can differentiate the prefetched ones from the predefined ones...

### Alternative ideas:
1. We could lean into `assignees` as being marked as predefined... and then make sure there is a max-height with scrolling for the "Tag Values" section.
2. Introduce something like `Tag.showAllValuesInAutocomplete: boolean` which we'd explicitly set for `is:`.
3. Even bigger UX change, but it would be nice to somehow condense `{attribute}`/ `un{attribute}` pairs to take up less space... but that's really not obvious to me.

2 seems reasonable to me and avoids making a very large chanfge if surfacing all of the `is:` options is desirable.

Reply:
Could filter keys define how many values they should show on an individual level? That would allow `is` to show more than 5 while assignee continues to show 5?

Reply:
@markstory : that would basically be similar to my 3 except there would be a "magic" number in the Tag, which could be beneficial if this list grows and we don't want to unboundedly show all of them:

```tsx
  getIssueAttributes() {
    // TODO(mitsuhiko): what do we do with translations here?
    return {
      is: {
        key: 'is',
        name: 'Status',
        values: ['resolved', 'unresolved', 'ignored', 'assigned', 'unassigned', 'linked', 'unlinked'],
        predefined: true,
        maxSearchItemsOverride: 7,
      },
```

Or if we wanted to ensure that for isValues we always show the full amount like my `showAllValuesInAutocomplete` solution (but can divorce them if we decide the list is getting long), I suppose we could write this as:

```tsx
  getIssueAttributes() {
    // TODO(mitsuhiko): what do we do with translations here?
    const isValues = ['resolved', 'unresolved', 'ignored', 'assigned', 'unassigned', 'linked', 'unlinked'];
    return {
      is: {
        key: 'is',
        name: 'Status',
        values: isValues,
        predefined: true,
        maxSearchItemsOverride: isValues.length,
      },
```

Thoughts on which is cleaner?

Reply:
Having a `maxSuggestedValues` instead of a `showAll` would be my preference. Using `.length` on a value list lets us get a 'show all' behavior.